### PR TITLE
Refactor/rx

### PR DESCRIPTION
**Pull request by DanielSilva20**

### CODE REVIEW

File: Challange-1-iOS/Challange-1-iOS/Network/NetworkManager.swift [hunk 1]
```diff
@@ -52,4 +55,31 @@ class NetworkManager {
 
         task.resume()
     }
+
+    func rxExecuteNetworkCall<ResultType: Decodable>(_ call: APIProtocol) -> Single<ResultType> {
+            let decoder = JSONDecoder()
```
Comment (cpradobliss) [inline Challange-1-iOS/Challange-1-iOS/Network/NetworkManager.swift]: It seems there are some extra spaces in the indentation here. Can you check?

File: Challange-1-iOS/Challange-1-iOS/Persistence/AvatarPersistence.swift [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import UIKit
 import CoreData
```
Comment (cpradobliss) [inline Challange-1-iOS/Challange-1-iOS/Persistence/AvatarPersistence.swift]: This empty line can be removed

**Comment by cpradobliss** [comment]
@DanielSilva20 nice PR in general! ⭐️

There are 2 points of improvement here:

- The indentation seems to be wrong in some places, please double-check that.
- You are calling your rx methods using the rx prefix in the method (e.g. `rxSomething`), a nice improvement here would be changing that syntax to: `rx.something`, how can this be done?

**Review by cpradobliss — CHANGES_REQUESTED** [review]

File: Challange-1-iOS/Challange-1-iOS/Persistence/AvatarPersistence.swift (hunk 2 above)
Comment (DanielSilva20) [reply Challange-1-iOS/Challange-1-iOS/Persistence/AvatarPersistence.swift]: done

File: Challange-1-iOS/Challange-1-iOS/Network/NetworkManager.swift (hunk 1 above)
Comment (DanielSilva20) [reply Challange-1-iOS/Challange-1-iOS/Network/NetworkManager.swift]: done

**Review by cpradobliss — APPROVED** [review]

**Pull request merged**